gaurav/src/pages/recipe/SearchResults.tsx

```diff
@@ -0,0 +1,164 @@
+import Input from "@/components/ui/Input";
+import axios from "axios";
+import { Search } from "lucide-react";
+import { useEffect, useState } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import RecipeCard from "./components/RecipeCard";
+import { Pagination } from "./components/Pagination";
+import { Toast } from "./components/Toast";
+import { IRecipeData } from "@/lib/interfaces";
+
+export const SearchResults = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const [searchresults, setSearchResults] = useState<IRecipeData[]>([]);
+  const [searchQuery, setSearchQuery] = useState<string>("");
+  const [savedRecipe, setSavedRecipe] = useState<IRecipeData[]>([]);
+  const [showToast, setShowToast] = useState(false);
+  const [toastMessage, setToastMessage] = useState("");
+  const [savedStatus, setSavedStatus] = useState<{ [key: string]: boolean }>({}); // {idMeal: true}
+  const [currentPage, setCurrentPage] = useState(1);
+  const [recipesPerPage, setRecipesPerPage] = useState(6);
+
+  const query = new URLSearchParams(location.search).get("query");
+
+  const API_URL = `https://www.themealdb.com/api/json/v1/1/search.php?s=${query}`;
```

Comment:
instead of declaring different API_URL variables in 3 places, declare at one and use it in all other places